cmd/adminkubeconfig/get.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure/auth"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/mgmt/redhatopenshift"
+	utillog "github.com/Azure/ARO-RP/pkg/util/log"
+)
+
+var (
+	gitCommit    = "unknown"
+	requiredEnvs = []string{
+		"AZURE_SUBSCRIPTION_ID", "AZURE_TENANT_ID",
+		"CLUSTER", "RESOURCEGROUP", "AZURE_CLIENT_ID", "AZURE_CLIENT_SECRET",
+	}
+)
+
+func usage() {
+	fmt.Fprintf(flag.CommandLine.Output(), "usage:\n")
+	for _, key := range requiredEnvs {
+		fmt.Fprintf(flag.CommandLine.Output(), "  %s=<value>\n", key)
+	}
+	fmt.Fprintf(flag.CommandLine.Output(), "  %s kubeconfig-destination\n", os.Args[0])
+	flag.PrintDefaults()
+}
+
+func writeKubeconfig(ctx context.Context, log *logrus.Entry, dest string) error {
+	conf := auth.NewClientCredentialsConfig(os.Getenv("AZURE_CLIENT_ID"), os.Getenv("AZURE_CLIENT_SECRET"), os.Getenv("AZURE_TENANT_ID"))
+	authorizer, err := conf.Authorizer()
+	if err != nil {
+		return err
+	}
+
+	openshiftclusters := redhatopenshift.NewOpenShiftClustersClient(os.Getenv("AZURE_SUBSCRIPTION_ID"), authorizer)
+
+	oc, err := openshiftclusters.Get(ctx, os.Getenv("RESOURCEGROUP"), os.Getenv("CLUSTER"))
```

Comment:
the contract for hack/db is that the caller puts the full cluster resource ID as os.Args[1] - I am tempted to use the same contract here and pick it apart here using azure.ParseResourceID.

The rationale for this is that generally I want us to try and think more in terms of resource IDs where possible instead of (subscriptionID, resourceGroupName, resourceID) - I think this will be especially useful when we get to geneva actions because it leads to less copy/pasting into fields horror.